File: app/src/main/java/com/example/tdm_android/activities/ProfileActivity.java
package com.example.tdm_android.activities;

import androidx.annotation.NonNull;
import androidx.appcompat.app.ActionBarDrawerToggle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.view.GravityCompat;
import androidx.drawerlayout.widget.DrawerLayout;

import android.app.Dialog;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toolbar;

import com.example.tdm_android.R;
import com.example.tdm_android.constants.Constants;
import com.example.tdm_android.managers.UserManager;
import com.example.tdm_android.models.User;
import com.google.android.material.navigation.NavigationView;

public class ProfileActivity extends AppCompatActivity {

    DrawerLayout drawerLayout;
    NavigationView navigationView;
    Toolbar toolbar;
    ActionBarDrawerToggle actionBarDrawerToggle;

    TextView tvUsername;
    TextView tvEmail;
    EditText etUsername;
    EditText etEmail;
    EditText etFullname;
    Button btnUpdateProfile;

    User userDetails;

    SharedPreferences pref;
    SharedPreferences.Editor editor;

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {

        if(actionBarDrawerToggle.onOptionsItemSelected(item)) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_profile);

        initializeVariables();

        btnUpdateProfile.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                updateUser(userDetails);
            }
        });

        drawerLayout = findViewById(R.id.drawer_layout);
        navigationView = findViewById(R.id.navigationView);
        actionBarDrawerToggle = new ActionBarDrawerToggle(this, drawerLayout, R.string.menu_open, R.string.menu_close);
        drawerLayout.addDrawerListener(actionBarDrawerToggle);
        actionBarDrawerToggle.syncState();
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        navigationView.setNavigationItemSelectedListener(new NavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {

                switch (item.getItemId()) {
                    case R.id.nav_home:
                        drawerLayout.closeDrawer(GravityCompat.START);
                        Intent intent = new Intent(ProfileActivity.this, FilterActivity.class);
                        startActivity(intent);
                        break;
                    case R.id.nav_favourites:
                        drawerLayout.closeDrawer(GravityCompat.START);
                        Intent intent1 = new Intent(ProfileActivity.this, FavouritesActivity.class);
                        startActivity(intent1);
                        break;
                    case R.id.nav_profile:
                        drawerLayout.closeDrawer(GravityCompat.START);
                        Intent intent2 = new Intent(ProfileActivity.this, ProfileActivity.class);
                        startActivity(intent2);
                        break;
                    case R.id.nav_logout:
                        drawerLayout.closeDrawer(GravityCompat.START);
                        Intent intent3 = new Intent(ProfileActivity.this, LoginActivity.class);
                        startActivity(intent3);
                        break;
                }

                return true;
            }
        });

    }

    private void initializeVariables() {

        tvUsername = (TextView) findViewById(R.id.tvUsername);
        tvEmail = (TextView) findViewById(R.id.tvEmail);
        etUsername = (EditText) findViewById(R.id.etUsername);
        etEmail = (EditText) findViewById(R.id.etEmail);
        etFullname = (EditText) findViewById(R.id.etUsername);
        btnUpdateProfile = (Button) findViewById(R.id.btnUpdateProfile);
        pref = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
        editor = pref.edit();

        userDetails = getUserDetails(this.pref.getString(Constants.STR_USERNAME, Constants.STR_USERNAME));

        tvUsername.setText(userDetails.getUsername());
        tvEmail.setText(userDetails.getEmail());
        etUsername.setText(userDetails.getUsername());
        etEmail.setText(userDetails.getEmail());
        etFullname.setText(userDetails.getUsername());

    }

    public User getUserDetails(String strUsername){
        try {
            User user = UserManager.getInstance(ProfileActivity.this).getOneUserByField("username", strUsername);
            return user;
        } catch (Exception e){
            e.printStackTrace();
        }
        return null;
    }

    public void updateUser(User user){

        userDetails.setUsername(String.valueOf(etUsername.getText()));
        tvUsername.setText(String.valueOf(etUsername.getText()));
        editor.putString(Constants.STR_USERNAME, etUsername.getText().toString());

        try {
            UserManager.getInstance(ProfileActivity.this).updateUser(user);
        } catch (Exception e){
            e.printStackTrace();
        }
    }

}

File: app/src/main/java/com/example/tdm_android/activities/ProfileActivity.kt
package com.example.tdm_android.activities

import android.annotation.SuppressLint
import com.example.tdm_android.R
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.drawerlayout.widget.DrawerLayout
import com.google.android.material.navigation.NavigationView
import androidx.appcompat.app.ActionBarDrawerToggle
import android.content.Intent
import android.content.SharedPreferences
import android.preference.PreferenceManager
import android.view.MenuItem
import android.view.View
import android.widget.*
import androidx.lifecycle.lifecycleScope
import com.example.tdm_android.constants.Constants
import com.example.tdm_android.functions.messageShort
import com.example.tdm_android.functions.triggerByChoosingNavigationMenuItem
import com.example.tdm_android.managers.UserManager
import com.example.tdm_android.models.User
import java.lang.Exception

class ProfileActivity : AppCompatActivity() {

    lateinit var drawerLayout: DrawerLayout
    lateinit var navigationView: NavigationView
    lateinit var actionBarDrawerToggle: ActionBarDrawerToggle

    lateinit var tvUsername: TextView
    lateinit var tvEmail: TextView
    lateinit var etUsername: EditText
    lateinit var etEmail: EditText
    lateinit var etFullname: EditText
    lateinit var btnUpdateProfile: Button
    lateinit var btnLogout: Button

    lateinit var userDetails: User
    lateinit var pref: SharedPreferences
    lateinit var editor: SharedPreferences.Editor

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return if (actionBarDrawerToggle.onOptionsItemSelected(item)) {
            true
        } else super.onOptionsItemSelected(item)
    }

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_profile)

        initializeVariables()

        btnUpdateProfile.setOnClickListener { updateUser(userDetails) }
        drawerLayout = findViewById(R.id.drawer_layout)
        navigationView = findViewById(R.id.navigationView)
        actionBarDrawerToggle = ActionBarDrawerToggle(this, drawerLayout, R.string.menu_open, R.string.menu_close)
        drawerLayout.addDrawerListener(actionBarDrawerToggle)
        actionBarDrawerToggle.syncState()
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)

        triggerByChoosingNavigationMenuItem(lifecycleScope, navigationView, drawerLayout)

        btnLogout.setOnClickListener {
            logoutUser()
            val intent = Intent(this@ProfileActivity, LoginActivity::class.java)
            startActivity(intent)
        }

    }

    @SuppressLint("CutPasteId")
    private fun initializeVariables() {
        tvUsername = findViewById<View>(R.id.tvUsername) as TextView
        tvEmail = findViewById<View>(R.id.tvEmail) as TextView
        etUsername = findViewById<View>(R.id.etUsername) as EditText
        etEmail = findViewById<View>(R.id.etEmail) as EditText
        etFullname = findViewById<View>(R.id.etUsername) as EditText
        btnUpdateProfile = findViewById<View>(R.id.btnUpdateProfile) as Button
        btnLogout = findViewById<View>(R.id.btnLogout) as Button
        pref = PreferenceManager.getDefaultSharedPreferences(applicationContext)
        editor = pref.edit()
        userDetails = getUserDetails(pref.getString(Constants.STR_USERNAME, Constants.STR_USERNAME))!!
        tvUsername.text = userDetails.username
        tvEmail.text = userDetails.email
        etUsername.setText(userDetails.username)
        etEmail.setText(userDetails.email)
        etFullname.setText(userDetails.username)
    }

    private fun getUserDetails(strUsername: String?): User? {
        try {
            return UserManager.getInstance(this@ProfileActivity)!!
                .getOneUserByField("username", strUsername)
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return null
    }

    private fun updateUser(user: User?) {

        userDetails.username = (etUsername.text.toString())

        messageShort(etUsername.text.toString())
        tvUsername.text = etUsername.text.toString()
        editor.putString(Constants.STR_USERNAME, etUsername.text.toString())
        editor.apply()

        try {
            UserManager.getInstance(this@ProfileActivity)!!
                .updateUser(user)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    private fun logoutUser() {
        pref = PreferenceManager.getDefaultSharedPreferences(applicationContext)
        editor = pref.edit()
        editor.putString(Constants.STR_USERNAME, "")
        editor.putString(Constants.STR_PASSWORD, "")
        editor.putBoolean(Constants.STR_CHECK_REMEMBER_USER, false)
        editor.apply()
    }

}